tests/pytests/test_vecsim.py

```diff
@@ -1771,50 +1771,54 @@ def test_rdb_memory_limit():
-        self.index_sizes = {'FLAT': 80000 * n_shards, 'HNSW': 10000 * n_shards, 'SVS-VAMANA': 10000 * n_shards}
+        # We need at least DEFAULT_BLOCK_SIZE at every shard, due to nature of hash slot distribution, we need a bit extra
+        # for that reason we multiply by 1.1
+        minimal_svs_index_size = int(DEFAULT_BLOCK_SIZE * 1.1 * n_shards)
+        self.index_sizes = {'FLAT': 100 * n_shards, 'HNSW': 100 * n_shards, 'SVS-VAMANA': minimal_svs_index_size}
         self.hybrid_modes = ['BATCHES', 'ADHOC_BF']
         self.dim = 10
         self.type = 'FLOAT32'
 
     def tearDown(self): # cleanup after each test
         self.env.flush()
 
-    def run_long_queries(self, n_vec, query_vec):
+    def run_timeout_tests(self, n_vec, query_vec):
+        small_k = 10
         # STANDARD KNN
-        large_k = 1000
         # run query with no timeout. should succeed.
-        res = self.env.cmd('FT.SEARCH', 'idx', '*=>[KNN $K @vector $vec_param]', 'NOCONTENT', 'LIMIT', 0, large_k,
-                                   'PARAMS', 4, 'K', large_k, 'vec_param', query_vec.tobytes(),
+        res = self.env.cmd('FT.SEARCH', 'idx', '*=>[KNN $K @vector $vec_param]', 'NOCONTENT', 'LIMIT', 0, n_vec,
+                                   'PARAMS', 4, 'K', small_k, 'vec_param', query_vec.tobytes(),
                                    'TIMEOUT', 0)
-        self.env.assertEqual(res[0], large_k)
-        # run query with 1 millisecond timeout. should fail.
-        self.env.expect(
-            'FT.SEARCH', 'idx', '*=>[KNN $K @vector $vec_param]',
-            'NOCONTENT', 'LIMIT', 0, n_vec, 'PARAMS', 4, 'K', n_vec,
-            'vec_param', query_vec.tobytes(), 'TIMEOUT', 1
-        ).error().contains('Timeout limit was reached')
-
-        # RANGE QUERY
-        # run query with 1 millisecond timeout. should fail.
-        self.env.expect('FT.SEARCH', 'idx', '@vector:[VECTOR_RANGE 10000 $vec_param]', 'NOCONTENT', 'LIMIT', 0, n_vec,
-                   'PARAMS', 2, 'vec_param', query_vec.tobytes(),
-                   'TIMEOUT', 1).error().contains('Timeout limit was reached')
-
-        # HYBRID MODES
-        # Add some dummy documents so `-dummy` won't be empty and optimized away.
-        with self.env.getClusterConnectionIfNeeded() as conn:
-            for i in range(n_vec + 1, n_vec + 5 * self.env.shardsCount):
-                conn.execute_command('HSET', i, 't', 'dummy')
-        for mode in self.hybrid_modes:
-            self.env.expect(
-                'FT.SEARCH', 'idx', '(-dummy)=>[KNN $K @vector $vec_param HYBRID_POLICY $hp]',
-                'NOCONTENT', 'LIMIT', 0, n_vec, 'PARAMS', 6, 'K', n_vec,
-                'vec_param', query_vec.tobytes(), 'hp', mode, 'TIMEOUT', 1
-            ).error().contains('Timeout limit was reached')
+        self.env.assertEqual(res[0], small_k)
+
+        # Enable VECSIM timeout to simulate timeout in the vecsim library
+        allShards_vecsimTimeoutEnable(self.env)
+        try:
```

Comment:
Just for safety? I don't see a reason today to expect an exception. Consider commenting on why we do it.